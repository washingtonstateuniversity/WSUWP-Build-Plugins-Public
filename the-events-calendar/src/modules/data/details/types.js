/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_DETAILS = `${ PREFIX_EVENTS_STORE }/SET_DETAILS`;
export const SET_DETAILS_POST_TYPE = `${ PREFIX_EVENTS_STORE }/SET_DETAILS_POST_TYPE`;
export const SET_DETAILS_IS_LOADING = `${ PREFIX_EVENTS_STORE }/SET_DETAILS_IS_LOADING`;
