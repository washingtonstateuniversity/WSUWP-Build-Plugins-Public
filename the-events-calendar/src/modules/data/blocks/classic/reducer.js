/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	detailsTitle: '',
	organizerTitle: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_CLASSIC_DETAILS_TITLE:
			return {
				...state,
				detailsTitle: action.payload.title,
			};

		case types.SET_CLASSIC_ORGANIZERS_TITLE:
			return {
				...state,
				organizerTitle: action.payload.title,
			};

		default:
			return state;
	}
};
