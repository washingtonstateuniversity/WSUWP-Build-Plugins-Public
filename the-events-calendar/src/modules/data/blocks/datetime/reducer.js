/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

export const DEFAULT_STATE = {
	start: momentUtil.toDateTime( defaultStartMoment ),
	end: momentUtil.toDateTime( defaultEndMoment ),
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguage: '',
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: date.FORMATS.TIMEZONE.string,
	timeZoneLabel: date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	showDateInput: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguage: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_LABEL:
			return {
				...state,
				timeZoneLabel: action.payload.label,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		case types.SET_DATE_INPUT_VISIBILITY:
			return {
				...state,
				showDateInput: action.payload.show,
			};
		default:
			return state;
	}
};
