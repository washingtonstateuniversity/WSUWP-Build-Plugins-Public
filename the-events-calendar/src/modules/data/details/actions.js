/**
 * Internal dependencies
 */
import * as types from './types';

export const enableDetailsIsLoading = ( id ) => ( {
	type: types.SET_DETAILS_IS_LOADING,
	payload: {
		id,
		isLoading: true,
	},
} );

export const disableDetailsIsLoading = ( id ) => ( {
	type: types.SET_DETAILS_IS_LOADING,
	payload: {
		id,
		isLoading: false,
	},
} );

export const setDetails = ( id, details ) => ( {
	type: types.SET_DETAILS,
	payload: {
		id,
		details,
	},
} );

export const setDetailsPostType = ( id, postType ) => ( {
	type: types.SET_DETAILS_POST_TYPE,
	payload: {
		id,
		postType,
	},
} );
