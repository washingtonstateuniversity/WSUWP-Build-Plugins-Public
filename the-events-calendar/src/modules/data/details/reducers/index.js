export { default as details } from './details';
