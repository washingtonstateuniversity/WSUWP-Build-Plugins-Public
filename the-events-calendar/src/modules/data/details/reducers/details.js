/**
 * Internal dependencies
 */
import { editor } from '@moderntribe/common/data';
import { types } from '@moderntribe/events/data/details';

export const DEFAULT_STATE = {
	isLoading: false,
	details: {},
	postType: editor.EVENT,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DETAILS:
			return {
				...state,
				details: action.payload.details,
			};
		case types.SET_DETAILS_POST_TYPE:
			return {
				...state,
				postType: action.payload.postType,
			};
		case types.SET_DETAILS_IS_LOADING:
			return {
				...state,
				isLoading: action.payload.isLoading,
			};
		default:
			return state;
	}
};
