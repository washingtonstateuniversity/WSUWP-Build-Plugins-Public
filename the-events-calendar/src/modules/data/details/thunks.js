/**
 * External dependencies
 */
import { isEmpty } from 'lodash';

/**
 * Internal dependencies
 */
import * as actions from './actions';
import * as selectors from './selectors';
import { middlewares } from '@moderntribe/common/store';

export const fetchDetails = ( id ) => ( dispatch, getState ) => {
	const state = getState();
	const props = { name: id };
	const isLoading = selectors.getIsLoading( state, props );
	const details = selectors.getDetails( state, props );

	if ( ! isEmpty( details ) || isLoading ) {
		return;
	}

	const postType = selectors.getPostType( state, props );
	const options = {
		path: `${ postType }/${ id }`,
		actions: {
			start: () => dispatch( actions.enableDetailsIsLoading( id ) ),
			success: ( { body } ) => {
				dispatch( actions.setDetails( id, body ) );
				dispatch( actions.disableDetailsIsLoading( id ) );
			},
			error: () => dispatch( actions.disableDetailsIsLoading( id ) ),
		},
	};

	dispatch( middlewares.request.actions.wpRequest( options ) );
};
