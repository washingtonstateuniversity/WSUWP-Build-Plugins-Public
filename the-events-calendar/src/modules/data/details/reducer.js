/**
 * Internal dependencies
 */
import * as types from './types';
import { details } from './reducers';

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.SET_DETAILS:
		case types.SET_DETAILS_IS_LOADING:
		case types.SET_DETAILS_POST_TYPE:
			return {
				...state,
				[ action.payload.id ]: details( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};
