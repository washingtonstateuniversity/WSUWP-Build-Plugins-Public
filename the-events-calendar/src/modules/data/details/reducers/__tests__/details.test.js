/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/details';
import reducer, { DEFAULT_STATE } from '@moderntribe/events/data/details/reducers/details';

describe( '[STORE] - details reducer', () => {
	it( 'Should return default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should set the details', () => {
		expect( reducer( DEFAULT_STATE, actions.setDetails( 20, { title: 'Modern Tribe' } ) ) )
			.toMatchSnapshot();
	} );

	it( 'Should enable the loading', () => {
		expect( reducer( DEFAULT_STATE, actions.enableDetailsIsLoading( 20 ) ) )
			.toMatchSnapshot();
	} );

	it( 'Should set the post type', () => {
		expect( reducer( DEFAULT_STATE, actions.setDetailsPostType( 20, 'tribe_organizers' ) ) )
			.toMatchSnapshot();
	} );
} );
