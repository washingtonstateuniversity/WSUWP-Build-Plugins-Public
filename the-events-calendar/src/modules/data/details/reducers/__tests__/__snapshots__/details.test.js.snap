// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - details reducer Should enable the loading 1`] = `
Object {
  "details": Object {},
  "isLoading": true,
  "postType": "tribe_events",
}
`;

exports[`[STORE] - details reducer Should set the details 1`] = `
Object {
  "details": Object {
    "title": "Modern Tribe",
  },
  "isLoading": false,
  "postType": "tribe_events",
}
`;

exports[`[STORE] - details reducer Should set the post type 1`] = `
Object {
  "details": Object {},
  "isLoading": false,
  "postType": "tribe_organizers",
}
`;
